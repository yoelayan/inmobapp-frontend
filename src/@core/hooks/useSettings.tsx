// React Imports
import { useContext } from 'react'

// Context Imports
import { SettingsContext } from '@core/contexts/settingsContext'

// Config Imports
import themeConfig from '@configs/themeConfig'
import primaryColorConfig from '@configs/primaryColorConfig'

export const useSettings = () => {
  // Hooks
  const context = useContext(SettingsContext)

  // During build or when context is not available, return default settings
  if (!context) {
    // Check if we're in a build environment or server-side rendering
    if (typeof window === 'undefined') {
      return {
        settings: {
          mode: themeConfig.mode,
          skin: themeConfig.skin,
          semiDark: themeConfig.semiDark,
          layout: themeConfig.layout,
          navbarContentWidth: themeConfig.navbar.contentWidth,
          contentWidth: themeConfig.contentWidth,
          footerContentWidth: themeConfig.footer.contentWidth,
          primaryColor: primaryColorConfig[0].main
        },
        updateSettings: () => {},
        isSettingsChanged: false,
        resetSettings: () => {},
        updatePageSettings: () => () => {}
      }
    }

    throw new Error('useSettingsContext must be used within a SettingsProvider')
  }

  return context
}
