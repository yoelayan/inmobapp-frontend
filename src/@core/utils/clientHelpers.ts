'use client'

// Type Imports
import type { Settings } from '@core/contexts/settingsContext'
import type { SystemMode } from '@core/types'

// Config Imports
import themeConfig from '@configs/themeConfig'

// Client-side implementations - for use only in Client Components
export const getSettingsFromCookieClient = (): Settings => {
  try {
    // Check if we're in a browser environment
    if (typeof window === 'undefined') {
      return {}
    }

    const cookieName = themeConfig.settingsCookieName

    // Try to get from localStorage first
    const localStorageSettings = localStorage.getItem(cookieName)

    if (localStorageSettings) {
      return JSON.parse(localStorageSettings)
    }

    // Fallback to document.cookie
    const cookies = document.cookie.split(';')
    const targetCookie = cookies.find(cookie => cookie.trim().startsWith(`${cookieName}=`))

    if (targetCookie) {
      const cookieValue = targetCookie.split('=')[1]

      return JSON.parse(decodeURIComponent(cookieValue))
    }

    return {}
  } catch (error) {
    console.error('Error getting settings from cookie on client:', error)

    return {}
  }
}

export const getModeClient = (): string => {
  const settingsCookie = getSettingsFromCookieClient()

  // Get mode from cookie or fallback to theme config
  return settingsCookie.mode || themeConfig.mode
}

export const getSystemModeClient = (): SystemMode => {
  const mode = getModeClient()

  // Check if we're in a browser environment
  if (typeof window === 'undefined') {
    return 'light'
  }

  // Get color preference from localStorage or cookie
  let colorPref = localStorage.getItem('colorPref') || 'light'

  // Check media query for dark mode preference if in system mode
  if (mode === 'system') {
    const prefersDarkMode = window.matchMedia('(prefers-color-scheme: dark)').matches

    colorPref = prefersDarkMode ? 'dark' : 'light'
  }

  return ((mode === 'system' ? colorPref : mode) as SystemMode) || 'light'
}

export const getServerModeClient = (): string => {
  const mode = getModeClient()
  const systemMode = getSystemModeClient()

  return mode === 'system' ? systemMode : mode
}

export const getSkinClient = (): string => {
  const settingsCookie = getSettingsFromCookieClient()

  return settingsCookie.skin || 'default'
}
