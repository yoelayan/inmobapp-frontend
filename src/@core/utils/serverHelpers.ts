// Next Imports
import { cookies } from 'next/headers'

// Type Imports
import type { Settings } from '@core/contexts/settingsContext'
import type { SystemMode } from '@core/types'

// Config Imports
import themeConfig from '@configs/themeConfig'

// Server-side implementations - these should only be used in Server Components
export const getSettingsFromCookie = async (): Promise<Settings> => {
  try {
    const cookieStore = await cookies()
    const cookieName = themeConfig.settingsCookieName
    const cookieValue = cookieStore.get(cookieName)?.value || '{}'

    return JSON.parse(cookieValue)
  } catch (error) {
    console.error('Error getting settings from cookie on server:', error)

    return {}
  }
}

export const getMode = async () => {
  const settingsCookie = await getSettingsFromCookie()

  // Get mode from cookie or fallback to theme config
  const _mode = settingsCookie.mode || themeConfig.mode

  return _mode
}

export const getSystemMode = async (): Promise<SystemMode> => {
  try {
    const cookieStore = await cookies()
    const mode = await getMode()
    const colorPrefCookie = (cookieStore.get('colorPref')?.value || 'light') as SystemMode

    return (mode === 'system' ? colorPrefCookie : mode) || 'light'
  } catch (error) {
    console.error('Error getting system mode on server:', error)

    return 'light'
  }
}

export const getServerMode = async () => {
  const mode = await getMode()
  const systemMode = await getSystemMode()

  return mode === 'system' ? systemMode : mode
}

export const getSkin = async () => {
  const settingsCookie = await getSettingsFromCookie()

  return settingsCookie.skin || 'default'
}
