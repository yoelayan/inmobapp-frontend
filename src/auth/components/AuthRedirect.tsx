'use client'

// Next Imports
import { useEffect } from 'react'

import { useRouter, usePathname } from 'next/navigation'

// Config Imports
import themeConfig from '@configs/themeConfig'

const AuthRedirect = () => {
  const pathname = usePathname()
  const router = useRouter()

  useEffect(() => {
    const redirectUrl = `/login?redirectTo=${pathname}`
    const login = `/login`
    const homePage = themeConfig.homePageUrl

    if (pathname !== login && pathname !== homePage) {
      router.push(redirectUrl)
    } else {
      router.push(login)
    }
  }, [pathname, router])

  return <></>
}

export default AuthRedirect
