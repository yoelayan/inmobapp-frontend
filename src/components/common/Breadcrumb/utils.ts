export interface BreadcrumbItem {
  label: string
  href?: string
  icon?: string
  badge?: string
}

// Route labels mapping for Spanish translations
const routeLabels: Record<string, string> = {
  // Main sections
  home: 'Inicio',
  about: 'Acerca de',
  clientes: 'Clientes',
  franquicias: 'Franquicias',
  propiedades: 'Propiedades',
  usuarios: 'Usuarios',
  roles: 'Roles',
  ejemplos: 'Ejemplos',

  // Actions
  crear: 'Crear',
  nuevo: 'Nuevo',
  agregar: 'Agregar',
  editar: 'Editar',
  ver: 'Ver',
  detalle: 'Detalle',
  detalles: 'Detalles',
  eliminar: 'Eliminar',
  buscar: 'Buscar',
  busquedas: 'Búsquedas',
  coincidencias: 'Coincidencias',
  matches: 'Coincidencias',

  // Generic terms
  lista: 'Lista',
  listado: 'Listado',
  formulario: 'Formulario',
  perfil: 'Perfil',
  configuracion: 'Configuración',
  configuraciones: 'Configuraciones',
  ajustes: 'Ajustes',

  // Status terms
  activo: 'Activo',
  inactivo: 'Inactivo',
  pendiente: 'Pendiente',
  completado: 'Completado'
}

// Route icons mapping
const routeIcons: Record<string, string> = {
  home: 'tabler-home',
  about: 'tabler-info-circle',
  clientes: 'tabler-users',
  franquicias: 'tabler-building-store',
  propiedades: 'tabler-home-2',
  usuarios: 'tabler-user',
  roles: 'tabler-user',
  ejemplos: 'tabler-code',

  // Actions
  crear: 'tabler-plus',
  nuevo: 'tabler-plus',
  agregar: 'tabler-plus',
  editar: 'tabler-edit',
  ver: 'tabler-eye',
  detalle: 'tabler-eye',
  detalles: 'tabler-eye',
  eliminar: 'tabler-trash',
  buscar: 'tabler-search',
  busquedas: 'tabler-search',
  coincidencias: 'tabler-target',
  matches: 'tabler-target',

  // Generic
  lista: 'tabler-list',
  listado: 'tabler-list',
  formulario: 'tabler-forms',
  perfil: 'tabler-user-circle',
  configuracion: 'tabler-settings',
  configuraciones: 'tabler-settings',
  ajustes: 'tabler-settings'
}

/**
 * Checks if a string represents a numeric ID
 */
const isNumericId = (segment: string): boolean => {
  return /^\d+$/.test(segment)
}

/**
 * Checks if a string represents a UUID
 */
const isUUID = (segment: string): boolean => {
  const uuidRegex = /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i

  return uuidRegex.test(segment)
}

/**
 * Gets the appropriate label for a route segment
 */
const getSegmentLabel = (segment: string, index: number, segments: string[]): string => {
  const lowerSegment = segment.toLowerCase()

  // Check if it's a known route label
  if (routeLabels[lowerSegment]) {
    return routeLabels[lowerSegment]
  }

  // Handle IDs (numeric or UUID)
  if (isNumericId(segment) || isUUID(segment)) {
    const prevSegment = segments[index - 1]?.toLowerCase()

    // Contextual ID labeling based on previous segment
    switch (prevSegment) {
      case 'clientes':
        return `Cliente #${segment}`
      case 'franquicias':
        return `Franquicia #${segment}`
      case 'propiedades':
        return `Propiedad #${segment}`
      case 'usuarios':
        return `Usuario #${segment}`
      case 'roles':
        return `Rol #${segment}`
      case 'busquedas':
        return `Búsqueda #${segment}`
      default:
        return `#${segment}`
    }
  }

  // Handle special segments that might contain underscores or hyphens
  const normalizedSegment = segment
    .replace(/[-_]/g, ' ')
    .split(' ')
    .map(word => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
    .join(' ')

  return normalizedSegment
}

/**
 * Gets the appropriate icon for a route segment
 */
const getSegmentIcon = (segment: string, index: number, segments: string[]): string | undefined => {
  const lowerSegment = segment.toLowerCase()

  // Check if it's a known route icon
  if (routeIcons[lowerSegment]) {
    return routeIcons[lowerSegment]
  }

  // Handle IDs - use parent segment icon
  if (isNumericId(segment) || isUUID(segment)) {
    const prevSegment = segments[index - 1]?.toLowerCase()

    if (prevSegment && routeIcons[prevSegment]) {
      return routeIcons[prevSegment]
    }
  }

  return undefined
}

/**
 * Builds the href for a breadcrumb item based on the segments up to a certain index
 */
const buildHref = (segments: string[], upToIndex: number): string => {
  const pathSegments = segments.slice(0, upToIndex + 1)

  return '/' + pathSegments.join('/')
}

/**
 * Determines if a breadcrumb item should be clickable
 */
const isClickable = (segment: string, index: number, segments: string[], isLast: boolean): boolean => {
  // Last item is never clickable (current page)
  if (isLast) {
    return false
  }

  // Don't make IDs clickable unless they lead to a view page
  if (isNumericId(segment) || isUUID(segment)) {
    // Check if next segment is a view action
    const nextSegment = segments[index + 1]?.toLowerCase()

    return nextSegment === 'ver' || nextSegment === 'detalle' || nextSegment === 'detalles'
  }

  // Make main sections clickable
  const mainSections = ['busquedas', 'clientes', 'franquicias', 'propiedades', 'usuarios', 'roles', 'home', 'about']

  return mainSections.includes(segment.toLowerCase())
}

/**
 * Adds context-specific badges to breadcrumb items
 */
const getBadge = (segment: string, index: number, segments: string[]): string | undefined => {
  const lowerSegment = segment.toLowerCase()

  // Add "Nuevo" badge for create actions
  if (['crear', 'nuevo', 'agregar'].includes(lowerSegment)) {
    return 'Nuevo'
  }

  // Add "Edit" badge for edit actions
  if (lowerSegment === 'editar') {
    return 'Editar'
  }

  // Add "Ver" badge for view actions
  if (['ver', 'detalle', 'detalles'].includes(lowerSegment)) {
    return 'Ver'
  }

  return undefined
}

/**
 * Main function to generate breadcrumb items from a pathname
 */
export const getBreadcrumbItems = (pathname: string): BreadcrumbItem[] => {
  // Remove leading slash and split into segments
  const segments = pathname.replace(/^\//, '').split('/').filter(Boolean)

  // Handle empty pathname (home page)
  if (segments.length === 0 || segments[0] === 'home') {
    return [{ label: 'Inicio', href: '/home', icon: 'tabler-home' }]
  }

  // Handle home page

  // Generate breadcrumb items
  const breadcrumbItems: BreadcrumbItem[] = segments.map((segment, index) => {
    const isLast = index === segments.length - 1
    const label = getSegmentLabel(segment, index, segments)
    const icon = getSegmentIcon(segment, index, segments)
    const href = isClickable(segment, index, segments, isLast) ? buildHref(segments, index) : undefined
    const badge = getBadge(segment, index, segments)

    return {
      label,
      href,
      icon,
      badge
    }
  })

  return breadcrumbItems
}

/**
 * Helper function to create custom breadcrumb items
 */
export const createBreadcrumbItem = (
  label: string,
  href?: string,
  icon?: string,
  badge?: string
): BreadcrumbItem => ({
  label,
  href,
  icon,
  badge
})

/**
 * Helper function to create breadcrumb items for common patterns
 */
export const createCommonBreadcrumbs = {
  home: () => createBreadcrumbItem('Inicio', '/home', 'tabler-home'),

  clients: () => createBreadcrumbItem('Clientes', '/clientes', 'tabler-users'),
  clientDetail: (id: string) => createBreadcrumbItem(`Cliente #${id}`, undefined, 'tabler-user'),
  clientEdit: (id: string) => createBreadcrumbItem('Editar', undefined, 'tabler-edit', 'Editar'),
  clientNew: () => createBreadcrumbItem('Nuevo Cliente', undefined, 'tabler-plus', 'Nuevo'),

  franchises: () => createBreadcrumbItem('Franquicias', '/franquicias', 'tabler-building-store'),
  franchiseDetail: (id: string) => createBreadcrumbItem(`Franquicia #${id}`, undefined, 'tabler-building-store'),
  franchiseEdit: (id: string) => createBreadcrumbItem('Editar', undefined, 'tabler-edit', 'Editar'),
  franchiseView: (id: string) => createBreadcrumbItem('Ver', undefined, 'tabler-eye', 'Ver'),

  properties: () => createBreadcrumbItem('Propiedades', '/propiedades', 'tabler-home-2'),
  propertyDetail: (id: string) => createBreadcrumbItem(`Propiedad #${id}`, undefined, 'tabler-home-2'),
  propertyEdit: (id: string) => createBreadcrumbItem('Editar', undefined, 'tabler-edit', 'Editar'),
  propertyNew: () => createBreadcrumbItem('Nueva Propiedad', undefined, 'tabler-plus', 'Nuevo'),

  users: () => createBreadcrumbItem('Usuarios', '/usuarios', 'tabler-user'),
  userDetail: (id: string) => createBreadcrumbItem(`Usuario #${id}`, undefined, 'tabler-user'),
  userEdit: (id: string) => createBreadcrumbItem('Editar', undefined, 'tabler-edit', 'Editar'),
  userView: (id: string) => createBreadcrumbItem('Ver', undefined, 'tabler-eye', 'Ver'),

  roles: () => createBreadcrumbItem('Roles', '/roles', 'tabler-user'),
  roleDetail: (id: string) => createBreadcrumbItem(`Rol #${id}`, undefined, 'tabler-user'),
  roleEdit: (id: string) => createBreadcrumbItem('Editar', undefined, 'tabler-edit', 'Editar'),
  roleView: (id: string) => createBreadcrumbItem('Ver', undefined, 'tabler-eye', 'Ver'),

  searches: () => createBreadcrumbItem('Búsquedas', '/busquedas', 'tabler-search'),
  searchDetail: (id: string) => createBreadcrumbItem(`Búsqueda #${id}`, undefined, 'tabler-search'),
  matches: () => createBreadcrumbItem('Coincidencias', undefined, 'tabler-target')
}
