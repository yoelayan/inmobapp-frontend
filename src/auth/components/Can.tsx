import type { ReactNode } from 'react'

import { useAuth } from '@auth/hooks/useAuth'

interface CanProps {
  permissions: string | string[]
  children: ReactNode
}

/**
 * Componente que renderiza condicionalmente contenido basado en los permisos del usuario
 * @param permissions - Un codename de permiso o array de codenames que el usuario debe tener
 * @param children - El contenido a renderizar si el usuario tiene los permisos
 * @example
 * <Can permissions="view_property">
 *   <PropertyList />
 * </Can>
 *
 * <Can permissions={['view_property', 'edit_property']}>
 *   <PropertyList />
 */
export const Can = ({ permissions, children }: CanProps) => {
  const { user } = useAuth()

  // Si no hay usuario autenticado, no renderizar nada
  if (!user || !user.user_permissions) {
    return null
  }

  // Normalizar permissions a array para facilitar la verificaciÃ³n
  const permissionsArray = Array.isArray(permissions) ? permissions : [permissions]

  // Verificar si el usuario tiene al menos uno de los permisos requeridos
  const hasPermission = permissionsArray.some(permission =>
    user.user_permissions.includes(permission)
  )

  // Solo renderizar children si el usuario tiene los permisos necesarios
  return hasPermission ? <>{children}</> : null
}

export default Can
