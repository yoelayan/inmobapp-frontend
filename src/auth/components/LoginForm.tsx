// React Imports
import React, { useState, useEffect } from 'react'



// MUI Imports
import { useRouter, useSearchParams } from 'next/navigation'

import IconButton from '@mui/material/IconButton'
import InputAdornment from '@mui/material/InputAdornment'
import Checkbox from '@mui/material/Checkbox'
import Button from '@mui/material/Button'
import FormControlLabel from '@mui/material/FormControlLabel'
import Typography from '@mui/material/Typography'

// API Imports
import ApiClient from '@/api/apiClient'


// Hooks Imports
import { useNotification } from '@/hooks/useNotification'
import { useAuth } from '@auth/hooks/useAuth'

// Component Imports
import Link from '@components/Link'
import CustomTextField from '@core/components/mui/TextField'

const LoginForm = () => {
  const authContext = useAuth()
  const { notify } = useNotification()
  const apiClient = ApiClient.getInstance()

  useEffect(() => {
    apiClient.setNotificationCallback(notify)
  }, [apiClient, notify])

  const { login } = authContext
  const [email, setEmail] = useState('')
  const [password, setPassword] = useState('')
  const [isPasswordShown, setIsPasswordShown] = useState(false)
  const searchParams = useSearchParams()
  const router = useRouter()

  const handleSubmit = async (e: React.FormEvent) => {
    e.preventDefault()
    await login(email, password)
    const redirectURL = searchParams.get('redirectTo') ?? '/'

    router.push(redirectURL)
  }

  const handleClickShowPassword = () => {
    setIsPasswordShown(!isPasswordShown)
  }

  return (
    <form noValidate autoComplete='off' onSubmit={handleSubmit} className='flex flex-col gap-5'>
      <CustomTextField
        autoFocus
        fullWidth
        label='Correo electrónico'
        placeholder='Ingresa tu correo electrónico'
        value={email}
        onChange={e => setEmail(e.target.value)}
        required
      />
      <CustomTextField
        fullWidth
        label='Contraseña'
        placeholder='············'
        id='outlined-adornment-password'
        type={isPasswordShown ? 'text' : 'password'}
        value={password}
        onChange={e => setPassword(e.target.value)}
        required
        InputProps={{
          endAdornment: (
            <InputAdornment position='end'>
              <IconButton edge='end' onClick={handleClickShowPassword} onMouseDown={e => e.preventDefault()}>
                <i className={isPasswordShown ? 'tabler-eye-off' : 'tabler-eye'} />
              </IconButton>
            </InputAdornment>
          )
        }}
      />
      <div className='flex justify-between items-center gap-x-3 gap-y-1 flex-wrap'>
        <FormControlLabel control={<Checkbox />} label='Remember me' />
        <Typography className='text-end' color='primary' component={Link}>
          Olvide mi contraseña
        </Typography>
      </div>
      <Button fullWidth variant='contained' type='submit'>
        Ingresar
      </Button>
    </form>
  )
}

export default LoginForm
