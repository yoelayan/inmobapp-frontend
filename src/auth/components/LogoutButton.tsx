// LogoutButton.jsx
import React from 'react'

import Button from '@mui/material/Button'

import { useAuth } from '@auth/hooks/useAuth'

const LogoutButton = () => {
  const authContext = useAuth()
  const logout = authContext?.logout || (() => {})

  const handleLogout = async () => {
    await logout()
  }

  return (
    <Button
      fullWidth
      variant='contained'
      color='error'
      size='small'
      endIcon={<i className='tabler-logout' />}
      onClick={handleLogout}
      sx={{ '& .MuiButton-endIcon': { marginInlineStart: 1.5 } }}
    >
      Logout
    </Button>
  )
}

export default LogoutButton
