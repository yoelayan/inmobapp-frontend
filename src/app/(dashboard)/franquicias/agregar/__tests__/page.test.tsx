/**
 * @jest-environment jsdom
 */
import React from 'react'

import { render, screen } from '@/tests/utils/test-utils'
import { setupHookMocks } from '@/tests/__mocks__/hooks'
import '@/tests/__mocks__/components'
import AddFranchise from '../page'

// Setup mocks
beforeEach(() => {
  setupHookMocks()
})

describe('Add Franchise Page', () => {
  it('should render without crashing', () => {
    render(<AddFranchise />)
    expect(screen.getByTestId('mock-sectionheader')).toBeInTheDocument()
    expect(screen.getByTestId('mock-franchiseform')).toBeInTheDocument()
  })

  it('should render section header with correct title', () => {
    render(<AddFranchise />)
    const sectionHeader = screen.getByTestId('mock-sectionheader')

    expect(sectionHeader).toBeInTheDocument()
  })

  it('should render franchise form component', () => {
    render(<AddFranchise />)
    expect(screen.getByTestId('mock-franchiseform')).toBeInTheDocument()
  })

  it('should have correct page structure', () => {
    const { container } = render(<AddFranchise />)

    expect(container.firstChild).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<AddFranchise />)

    expect(container.firstChild).toMatchSnapshot()
  })
})

// Test with different states
describe('Add Franchise Page - Different States', () => {
  it('should render when form is in loading state', () => {
    render(<AddFranchise />)
    expect(screen.getByTestId('mock-franchiseform')).toBeInTheDocument()
  })

  it('should render all required components', () => {
    render(<AddFranchise />)
    expect(screen.getByTestId('mock-sectionheader')).toBeInTheDocument()
    expect(screen.getByTestId('mock-franchiseform')).toBeInTheDocument()
  })

  it('should render components correctly', () => {
    const { container } = render(<AddFranchise />)


    // Just check that the container has content
    expect(container.firstChild).toBeInTheDocument()
    expect(container.firstChild).not.toBeEmptyDOMElement()
  })
})

// Test form integration
describe('Add Franchise Page - Form Integration', () => {
  it('should pass correct props to franchise form', () => {
    render(<AddFranchise />)
    const franchiseForm = screen.getByTestId('mock-franchiseform')

    expect(franchiseForm).toBeInTheDocument()
  })

  it('should handle form submission', () => {
    render(<AddFranchise />)
    const franchiseForm = screen.getByTestId('mock-franchiseform')

    expect(franchiseForm).toBeInTheDocument()

    // Form submission logic would be tested in the actual form component
  })

  it('should handle form errors', () => {
    render(<AddFranchise />)
    const franchiseForm = screen.getByTestId('mock-franchiseform')

    expect(franchiseForm).toBeInTheDocument()

    // Error handling would be tested in the actual form component
  })
})
