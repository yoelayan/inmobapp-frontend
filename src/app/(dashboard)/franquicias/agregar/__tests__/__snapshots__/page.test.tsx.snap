// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Franchise Page should match snapshot 1`] = `
<div
  data-theme-provider="true"
>
  <div
    data-testid="mock-sectionheader"
    subtitle="Crear una nueva franquicia en el sistema"
    title="Agregar Franquicia"
  />
  <div
    data-testid="mock-franchiseform"
  />
</div>
`;
